tests/v2/validation/provisioning/rke2/etcd_backup_restore_test.go

```diff
@@ -56,612 +41,117 @@ func (r *RKE2EtcdSnapshotRestoreTestSuite) SetupSuite() {
-
-	logrus.Infof("creating rke2Cluster.............")
-	clusterResp, err := createRKE2NodeDriverCluster(client, provider, clusterName, initialK8sVersion, r.ns, r.cnis[0], r.advancedOptions, r.etcdSnapshotS3)
-	require.NoError(r.T(), err)
-	require.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
-	logrus.Infof("rke2Cluster create request successful.............")
-
-	if r.client.Flags.GetValue(environmentflag.UpdateClusterName) {
-		pipeline.UpdateConfigClusterName(clusterName)
-	}
-
-	logrus.Infof("creating watch over cluster.............")
-	clusters.WatchAndWaitForCluster(r.client.Steve, kubeProvisioningClient, r.ns, clusterName)
-	logrus.Infof("cluster is up and running.............")
-
-	// Get clusterID by clusterName
-	logrus.Info("getting cluster id.............")
-	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
-	require.NoError(r.T(), err)
-	logrus.Info("got cluster id.............", clusterID)
-
-	logrus.Info("getting local cluster id.............")
-	localClusterID, err := clusters.GetClusterIDByName(client, localClusterName)
-	require.NoError(r.T(), err)
-	logrus.Info("got local cluster id.............", localClusterID)
-
-	logrus.Infof("creating watch over pods.............")
-	r.watchAndWaitForPods(client, clusterID)
-	logrus.Infof("All pods are up and running.............")
-
-	// creating the workload W1
-	logrus.Infof("creating a workload(nginx deployment).............")
-
-	containerTemplate := workloads.NewContainer("ngnix", "nginx", v1.PullAlways, []v1.VolumeMount{}, []v1.EnvFromSource{})
-	podTemplate := workloads.NewPodTemplate([]v1.Container{containerTemplate}, []v1.Volume{}, []v1.LocalObjectReference{}, nil)
-	deploymentBeforeBackup := workloads.NewDeploymentTemplate(wloadBeforeRestore, r.ns, podTemplate, isCattleLabeled, nil)
-
-	// creating steve client
-	steveclient, err := client.Steve.ProxyDownstream(clusterID)
-	require.NoError(r.T(), err)
-
-	deploymentResp, err := createDeployment(deploymentBeforeBackup, steveclient, client, clusterID)
-	require.NoError(r.T(), err)
-	require.Equal(r.T(), deploymentBeforeBackup.Name, deploymentResp.ObjectMeta.Name)
-	logrus.Infof("%v is ready.............", deploymentBeforeBackup.Name)
-
-	// creating the ingress1
-	logrus.Infof("creating an ingress.............")
-
-	exactPath := networkingv1.PathTypeExact
-	paths := []networkingv1.HTTPIngressPath{
+func (r *RKE2EtcdSnapshotRestoreTestSuite) TestEtcdOnlySnapshotRestore() {
+	logrus.Infof("checking for valid k8s versions and cnis in the configuration....")
+	require.GreaterOrEqualf(r.T(), len(r.clustersConfig.Basics.RKE2KubernetesVersions), 1, "At least one k8s version is required in the config")
+	require.GreaterOrEqualf(r.T(), len(r.clustersConfig.Basics.CNIs), 1, "At least one cni is required in the config")
+	nodeRoles := []machinepools.NodeRoles{
+		{
+			ControlPlane: true,
+			Quantity:     2,
+		},
+		{
+			Etcd:     true,
+			Quantity: 3,
+		},
 		{
-			Path:     "/index.html",
```

Comment:
I don't see where you put this. Was it purposely removed?

Reply:
it exists in crud/restore.go  -> createIngress:NewIngressPathTemplate as a 'new' function. I think the existing test file just didn't use this function everywhere that it could have.